### PR TITLE
[CI] Fix the CI!

### PR DESCRIPTION
**Pull request by erikwijmans**
## Motivation and Context

For some reason our CI started having certificate issues with cmake. AFAIK this is due to Ubuntu 16.04 being EOL. This PR moves the CI to 20.04.

There was one change needed to make things run on 20.04, which was make the method we call in the Simulator constructor not virtual (as clang-tidy rightfully complains about that being a can of worms). We currently don't overload those anywhere expect in python, but python doesn't need them to be marked as virtual.

## How Has This Been Tested

With the CI

 Docs change / refactoring / dependency upgrade

### CODE REVIEW

**Comment by Skylion007** [comment]
The some reason is that Let's Encrypt old Root CA expired this weekend: https://portswigger.net/daily-swig/lets-encrypt-root-cert-update-catches-out-many-big-name-tech-firms

**Review by Skylion007 — APPROVED** [review]

**Comment by erikwijmans** [comment]
Interesting. I wonder why that only seems to matter for the ubuntu 16.04 image.

File: tests/test_physics.py [hunk 1]
```diff
@@ -1710,7 +1710,7 @@ def test_rigid_constraints():
             habitat_sim.physics.RigidConstraintType.PointToPoint
         )
         constraint_id_2 = sim.create_rigid_constraint(constraint_settings_2)
-        observations += simulate(sim, 5.0, produce_debug_video)
+        observations += simulate(sim, 7.0, produce_debug_video)
```
Comment (erikwijmans) [inline tests/test_physics.py:1713]: @aclegg3 Due to compiler changes the assert on L1719 was failing. Cranking up situation time seemed to fix that.

**Comment by Skylion007** [comment]
Because the distro is so old, the Root CAs were never updated.

File: .circleci/config.yml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 version: 2.1
 gpu: &gpu
   machine:
-    image: ubuntu-1604:201903-01
+    image: ubuntu-2004:202107-02
```
Comment (Skylion007) [inline .circleci/config.yml:4]:
```suggestion
    image: ubuntu-2004-cuda-11.2:202103-01
```
This has CUDA 11.2 pre-installed which should simplify the installation if that is accessible. I don't know if a Ubuntu-2004 has a cuda 11.1 preinstalled: https://circleci.com/docs/2.0/configuration-reference/

**Comment by Skylion007** [comment]
LGTM. Let's merge this as it would be great if we can restore the CI. 👍

**Comment by Skylion007** [comment]
@erikwijmans I found a lot less invasive fix by doing `apt-get update/ apt-get upgrade` at the beginning of the circleci file and tested in the fix_rootca_quickfix branch.

File: .circleci/config.yml (hunk 2 above)
Comment (erikwijmans) [reply .circleci/config.yml:4]: I guess cuda-11.2 may be fine here, but in general cuda-11.2 is avoided since its a buggy slow mess.

**Comment by Skylion007** [comment]
@erikwijmans Alternative quickfix: https://github.com/facebookresearch/habitat-sim/pull/1518

**Comment by aclegg3** [comment]
Covered Ubuntu upgrade in #1665.

**Pull request closed without merging**